dungeon/test/manual/quizquestion/WizardQuizTest.java

```diff
@@ -0,0 +1,135 @@
+package manual.quizquestion;
+
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.ui.CheckBox;
+import com.badlogic.gdx.scenes.scene2d.ui.ScrollPane;
+import com.badlogic.gdx.scenes.scene2d.ui.TextArea;
+import com.badlogic.gdx.scenes.scene2d.ui.VerticalGroup;
+import com.badlogic.gdx.utils.SnapshotArray;
+import contrib.components.InteractionComponent;
+import contrib.configuration.ItemConfig;
+import contrib.entities.EntityFactory;
+import contrib.systems.*;
+
+import core.Entity;
+import core.Game;
+import core.components.DrawComponent;
+import core.components.PositionComponent;
+import core.components.UIComponent;
+import core.hud.UITools;
+import core.level.utils.LevelSize;
+import core.systems.LevelSystem;
+import task.Task;
+import task.TaskComponent;
+import task.quizquestion.QuizQuestion;
+import task.quizquestion.QuizUI;
+
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+import static task.quizquestion.QuizDialogDesign.ANSWERS_GROUP_NAME;
+
+public class WizardQuizTest {
+
+
+    public static QuizQuestion singleChoiceDummy() {
+        QuizQuestion question =
+            new QuizQuestion(
+                QuizQuestion.QuizType.SINGLE_CHOICE,
+                QuizQuestion.QuizContentType.TEXT,
+                "Was ist kein Ziel von Refactoring?");
+        question.addAnswer(QuizQuestion.QuizContentType.TEXT, "Lesbarkeit von Code verbessern");
+        question.addAnswer(
+            QuizQuestion.QuizContentType.TEXT, "Verständlichkeit von Code verbessern");
+        question.addAnswer(
+            QuizQuestion.QuizContentType.TEXT, "Wartbarkeit von Code verbessern");
+        question.addAnswer(QuizQuestion.QuizContentType.TEXT, "Fehler im Code ausmerzen");
+        return question;
+    }
+
+
+    private static final Consumer<Entity> wizardInteractionCallback =
+        entity -> {
+            Task quest = entity.fetch(TaskComponent.class).orElseThrow().task();
+            // build GUI with its callback
+            QuizUI.showQuizDialog((QuizQuestion) quest, (Entity wizard)->{
+            System.out.println("DEBUG WIZARD: 58" );
+                // TODO answer=
+                return (textDialog, id) -> {
+                    if (Objects.equals(id, core.hud.UITools.DEFAULT_DIALOG_CONFIRM)) {
+                        SnapshotArray<Actor> children = textDialog.getContentTable().getChildren();
```

Comment:
Ok, an der stelle bekomem ich eine VerticalGroup wieder, dessen kindern will ich eigentlich betrachten.

Also irgendwas in richtung ` children = textDialog.getContentTable().get(0).getChildren()`;

Reply:
```suggestion
                        SnapshotArray<Actor> children = ((VerticalGroup)textDialog.getContentTable().getChildren().get(0)).getChildren();
```

Hier ist eine ebene Verloren gegangen. ggfls ist es auch besser anstelle von hier eine suche nach den Elementen zu machen diese irgendwo zwischenspeichern und beim callback direkt zu verwenden. ob nun einfach eine Liste oder eine Map oder so. Dies gehört aber würde ich sagen zu dem Aspekt besser machen.